### DISCUSSION TITLE
Is there any way to render html (SSR) in apollo-server??

### DISCUSSION THREAD

**Discussion by Jsunmin** [apollographql/apollo-server#4906] [Q&A]
Hi, I want to render simple html page(SSR) for changing user password in Apollo-server, without Apollo-client.

I found some ways to do this.
 1, using apollo-server-express, and add middleware regarding view engine setting
   --> but I want to use apollo-server lib.
 2, with nextjs
   --> but it become too heavy, just for a html page..

is there any nice solution to do this??... 

Thanks

**Comment by glasser** [comment]
By `apollo-server`, you mean specifically the `apollo-server` package, not the other packages here?

The entire point of `apollo-server` is that it's really easy to set up if you don't need to do anything non-standard with the web server. If you want to do anything at all with the web server other than the defaults, just use `apollo-server-express`.  Using `apollo-server-express` is barely more work than `apollo-server`; the [entire source code of `apollo-server`](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server/src/index.ts) is 141 lines including imports and such, and the biggest part of that file is just calculating your app's URL to return from `listen()`.
